frontend/taipy-gui/src/components/Taipy/Metric.tsx

```diff
@@ -66,6 +67,22 @@ const Metric = (props: MetricProps) => {
     const hover = useDynamicProperty(props.hoverText, props.defaultHoverText, undefined);
     const theme = useTheme();
 
+    const colorMap = useMemo(() => {
+        try {
+            const obj = props.colorMap ? JSON.parse(props.colorMap) : null;
+            if (obj && typeof obj === 'object') {
+                const keys = Object.keys(obj);
+                return keys.sort((a, b) => Number(a) - Number(b)).map((key, index) => {
+                    const nextKey = keys[index + 1] !== undefined ? Number(keys[index + 1]) : props.max || 100;
```

Comment:
is it a problem when the max (100) is specified as the last key ?

Reply:
This shouldn't be an issue. If the last key is set to 100 without the "max" attribute defined, the color range will be [100,100]. This still defines the color for this range, but from the UI view, the user cannot see it because they overlap each other.

Reply:
so no big deal?

Reply:
no big deal at all 😎